clarifai/rag.py

```diff
@@ -0,0 +1,122 @@
+        >>> rag_agent = RAG()
+    """
+
+  def __init__(self,
+               workflow_url: str = None,
+               base_url: str = "https://api.clarifai.com",
+               pat: str = None,
+               **kwargs):
+    """
+        Initialize an empty or existing RAG.
+        """
+    w = Workflow(url=workflow_url, base_url=base_url, pat=pat)
+    self._prompt_workflow = w
+    self._app = App(app_id=w.app_id, base_url=w.base, pat=w.pat)
+    del w
+
+  def setup(self,
+            user_id: str = None,
+            llm_url: str = "https://clarifai.com/mistralai/completion/models/mistral-7B-Instruct",
+            base_workflow: str = "Text",
+            workflow_yaml_filename: str = 'prompter_wf.yaml',
+            base_url: str = "https://api.clarifai.com",
+            pat: str = None,
+            **kwargs):
+    """
+        Creates an app with `Text` as base workflow, create prompt model, create prompt workflow.
+        Example:
+        >>> from clarifai.rag import RAG
+        >>> rag_agent = RAG().setup()
+        """
+    user = User(user_id=user_id, base_url=base_url, pat=pat)
+    llm = Model(llm_url)
+
+    params = Struct()
+    params.update({
+        "prompt_template":
+            "Context information is below:\n{data.hits}\nGiven the context information and not prior knowledge, answer the query.\nQuery: {data.text.raw}\nAnswer: "
+    })
+    prompter_model_params = {"params": params}
+
+    ## Create an App
+    now_ts = str(int(datetime.now().timestamp()))
+    app_id = f"rag_app_{now_ts}"
+    app = user.create_app(app_id=app_id, base_workflow=base_workflow)
+
+    ## Create rag-prompter model and version
+    prompter_model = app.create_model(
+        model_id=f"rag_prompter_{now_ts}", model_type_id="rag-prompter")
+    prompter_model = prompter_model.create_version(output_info=prompter_model_params)
+
+    ## Generate a tmp yaml file for workflow creation
+    workflow_id = f"rag-wf-{now_ts}"
+    workflow_dict = {
+        "workflow": {
+            "id":
+                workflow_id,
+            "nodes": [{
+                "id": "rag-prompter",
+                "model": {
+                    "model_id": prompter_model.id,
+                    "model_version_id": prompter_model.model_version.id
+                }
+            }, {
+                "id": "llm",
+                "model": {
+                    "model_id": llm.id,
+                    "model_version_id": prompter_model.model_version.id
+                },
+                "node_inputs": [{
+                    "node_id": "rag-prompter"
+                }]
+            }]
+        }
+    }
+    with open(workflow_yaml_filename, 'w') as out_file:
+      yaml.dump(workflow_dict, out_file, default_flow_style=False)
+
+    ## Create prompt workflow
+    self._prompt_workflow = app.create_workflow(config_filepath=workflow_yaml_filename)
+    self._app = app
+
+    del user, llm, prompter_model, prompter_model_params
```

Comment:
Maybe add a helper function in [urls](https://github.com/Clarifai/clarifai-python/blob/master/clarifai/urls/helper.py), that generates urls for resource instance.

workflow_url = ClarifaiURLHelper.get_clarifai_url(self._prompt_workflow)

```suggestion
    return cls(workflow_url, base_url, pat)
```

Reply:
This assumes `setup` to be a classmethod. Let's merge the conversation to just the one above.